app/app.py

```diff
@@ -31,5 +86,44 @@ def get_prices(list):
         print(dict["dataset"]["data"][0][1])
 
 
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    form = LoginForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(username=form.username.data).first()
+        if user:
+            if bcrypt.check_password_hash(user.password, form.password.data):
+                login_user(user)
+                return redirect(url_for('data_summary.html'))
+    return render_template('login.html', form=form)
```

Comment:
I'm not sure the authentication is working properly for me ... After making the changes I mentioned above, I'm rerouted to the data summary page every time, regardless of what mumbo jumbo I include for username and password.